apps/web/src/design-system/template-button/ChannelButton.tsx

```diff
@@ -164,100 +164,149 @@ export function ChannelButton({
-                      setActivePage(tabKey === ChannelTypeEnum.IN_APP ? tabKey : capitalize(channelKey));
+                      onDelete(id || '');
                     }}
                   >
-                    Edit Template
+                    Delete {isChannel ? 'Step' : 'Action'}
                   </Menu.Item>
-                </When>
-                <Menu.Item
-                  key="delete"
-                  style={{
-                    pointerEvents: 'all',
-                  }}
-                  icon={<Trash />}
-                  data-test-id="delete-step-action"
-                  onClick={() => {
-                    setShowDotMenu(false);
-                    onDelete(id || '');
-                  }}
-                >
-                  Delete {isChannel ? 'Step' : 'Action'}
-                </Menu.Item>
-              </Menu.Dropdown>
-            </Menu>
-          </When>
-        </ActionWrapper>
-      </ButtonWrapper>
+                </Menu.Dropdown>
+              </Menu>
+            </When>
+          </ActionWrapper>
+        </ButtonWrapper>
+        {!hasActiveIntegration && (
+          <Popover
+            opened={popoverOpened}
+            withinPortal
+            transition="rotate-left"
+            transitionDuration={250}
+            offset={10}
+            target={<ErrorCircle data-test-id="error-circle" dark={theme.colorScheme === 'dark'} />}
+            title="Connect provider"
+            titleGradient="red"
+            description="Please configure a chat provider to send notifications over this channel"
+            content={
+              <ConfigureProviderButton
+                onClick={() => {
+                  setIntegrationsModalVisible(true);
+                  setPopoverOpened(false);
+                }}
+              >
+                Configure
+              </ConfigureProviderButton>
+            }
+          />
+        )}
+        {hasActiveIntegration && stepErrorContent && (
+          <MantinePopover
+            withinPortal
+            classNames={popoverClasses}
+            withArrow
+            opened={popoverOpened && Object.keys(stepErrorContent).length > 0}
+            transition="rotate-left"
+            transitionDuration={250}
+            offset={theme.spacing.xs}
+            position="right"
+            zIndex={4}
+            positionDependencies={[dragging, viewport]}
+            clickOutsideEvents={MENU_CLICK_OUTSIDE_EVENTS}
+          >
+            <MantinePopover.Target>
+              <ErrorCircle data-test-id="error-circle" dark={theme.colorScheme === 'dark'} />
+            </MantinePopover.Target>
+            <MantinePopover.Dropdown>
+              <Text rows={1} color={colors.white}>
+                {stepErrorContent || 'Something is missing here'}
+              </Text>
+            </MantinePopover.Dropdown>
+          </MantinePopover>
+        )}
+      </UnstyledButtonStyled>
       <IntegrationsStoreModal
```

Comment:
moved the modal outside, because we don't want receive the mouse events on the node button when hovering over the modal